### PR TITLE
More configurable rpmconf

### PR DESCRIPTION
**Pull request by mcspr**
- add /etc/dnf/plugins/rpmconf.conf
- can configure rpmconf frontend (partly RhBug [990444](https://bugzilla.redhat.com/show_bug.cgi?id=990444))
- can configure rpmconf to simply display diffs
- disable plugin with `dnf --assume[yes|no]` / no tty \(RhBugs [1211344](https://bugzilla.redhat.com/show_bug.cgi?id=1211344) and [1213848](https://bugzilla.redhat.com/show_bug.cgi?id=1213848)\)
- catch sys.exit calls from rpmconf so it doesn't quit dnf
- tests for all of the above

read_config call in plugins/rpm_conf.py requires rpm-software-management/dnf#492

### CODE REVIEW

**Comment by ignatenkobrain** [comment]
We just merged some enhancement on reading conf file (comes with DNF 2.0 API), please change according to it as well.

File: plugins/rpm_conf.py [hunk 1]
```diff
@@ -30,8 +31,32 @@ def __init__(self, base, cli):
...
+        self.diff = None
+
+    def config(self):
+        self._interactive = True
+        if (not sys.stdin or not sys.stdin.isatty()) \
+                or self.base.conf.assumeyes \
+                or self.base.conf.assumeno:
+            self._interactive = False
+
+        conf = self.read_config(self.base.conf)
```
Comment (mcspr) [inline plugins/rpm_conf.py]: @ignatenkobrain If you mean #58, this should already do it.

Comment (ignatenkobrain) [inline plugins/rpm_conf.py]: ah, right. didn't see that.

File: plugins/rpm_conf.py [hunk 2]
```diff
@@ -43,5 +68,23 @@ def resolved(self):
...
+            return
+
+        rconf = rpmconf.RpmConf(
+            packages=self.packages,
+            frontend=self.frontend,
+            diff=self.diff)
+        try:
+            rconf.run()
+        except SystemExit as e:
+            if e.code == 2:
```
Comment (ignatenkobrain) [inline plugins/rpm_conf.py]: please no magic numbers ;) there should be some -EERROR ;)

Comment (mcspr) [inline plugins/rpm_conf.py]: by the way [rpmconf module got them as numbers] (https://github.com/xsuchy/rpmconf/blob/f0543ced24227f5bb2ebe591278ae27c5a8d34ce/rpmconf/rpmconf.py#L275-L281)

**Pull request merged**